### PR TITLE
External VAD

### PR DESCRIPTION
**Pull request by AudranBert**
- Can send speech segments to faster whisper (allow using external VAD)

### CODE REVIEW

File: faster_whisper/transcribe.py [hunk 1]
```diff
@@ -318,8 +318,11 @@ def transcribe(
         self.logger.info(
             "Processing audio with duration %s", format_timestamp(duration)
         )
-
-        if vad_filter:
+        if isinstance(vad_filter, list):
```
Comment (Jeronymous) [inline faster_whisper/transcribe.py]: Given that you expect an iterable, can we generalize this test to support any kinds of iterable object?

File: faster_whisper/transcribe.py [hunk 2]
```diff
@@ -231,7 +231,7 @@ def transcribe(
         word_timestamps: bool = False,
         prepend_punctuations: str = "\"'“¿([{-",
         append_punctuations: str = "\"'.。,，!！?？:：”)]}、",
-        vad_filter: bool = False,
+        vad_filter: Union[bool, Iterable] = False,
```
Comment (Jeronymous) [inline faster_whisper/transcribe.py]: Please update the doc string for "vad_filter" also.
In particular, I guess there is an expected format, like either [(start, end), ...] or [{"start": start, "end": end}]. Please describe this expected format somewhere.

File: faster_whisper/transcribe.py [hunk 3]
```diff
@@ -318,8 +318,11 @@ def transcribe(
         self.logger.info(
             "Processing audio with duration %s", format_timestamp(duration)
         )
-
-        if vad_filter:
+        if isinstance(vad_filter, list):
+            speech_chunks = vad_filter
+            audio = collect_chunks(audio, speech_chunks)
+            duration_after_vad = audio.shape[0] / sampling_rate
```
Comment (Jeronymous) [inline faster_whisper/transcribe.py]: These lines are copy-pasted. It means that the code could be factored (while trying to keep the change minimal)
For instance:
```
if vad_filter:
   ... 
   if isinstance(vad_filter, bool):
      speech_chunks = get_speech_timestamps(audio, vad_parameters)
   else:
      speech_chunks = vad_filter # maybe with some checks...
   ... after the same happens ...
```

**Review by Jeronymous — COMMENTED** [review]
Don't forget to test what happens in the corner case where the passed speech segments are an empty list.

**Comment by AudranBert** [comment]
> Don't forget to test what happens in the corner case where the passed speech segments are an empty list.

It will be the same thing as vad_filter=False meaning it will process the whole audio
